albumentations/core/composition.py

```diff
@@ -269,6 +269,61 @@ def __call__(self, force_apply=False, **data):
         return data
 
 
+class SomeOf(BaseCompose):
+    """Select some of the transforms to apply. Selected transforms will be called with `force_apply=True`.
+    Transforms probabilities will be normalized to one 1, so in this case transforms probabilities works as weights.
+
+    Args:
+        sample_range (int, int): Range of possible sample sizes.
+        transforms (list): list of transformations to compose.
+        p (float): probability of applying selected transform. Default: 0.5.
+    """
+
+    def __init__(self, sample_range: (int, int), transforms, p=0.5):
+        super().__init__(transforms, p)
+        if isinstance(sample_range, (list, tuple)):
+            self.sample_range = sample_range
+        elif isinstance(sample_range, int):
+            self.sample_range = (0, sample_range)
+        else:
+            raise TypeError(f'Invalid type for sample_range: {type(sample_range).__name__}')
+        self.sample_range = sample_range
+        assert len(self.sample_range) == 2
```

Comment:
We use exceptions instead of assertions to check input arguments for validity. We do use assertions in unit tests.

Reply:
Fixed in latest commit